.github/workflows/codspeed.yml

```diff
@@ -0,0 +1,59 @@
+name: CodSpeed Benchmarks
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - 'justfile'
+      - '.devcontainer'
+      - '**/*.md'
+      - '.gitignore'
+  merge_group:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: codspeed.yml-${{ github.event.pull_request.number || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  benchmarks:
+    name: Run benchmarks
+    runs-on: ubuntu-22.04
+    env:
+      BAZEL_ARGS: --config=ci --config=ci-linux --@codspeed_cpp//core:codspeed_mode=instrumentation --compilation_mode=dbg --copt=-O2 --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev
```

Comment:
Should we add --config=benchmark to here as well?

Reply:
We should not be using dbg here or specify -O2 manually. Based on the configurations we currently have, `--config=benchmark` is a workable choice here, although this doesn't exactly give us full optimizations (but that's another topic, should be addressed in a wider bazelrc refactor)